libres/tests/analysis/test_update.cpp

```diff
@@ -0,0 +1,120 @@
+
+SCENARIO("Running analysis update without row scaling on linear model",
+         "[analysis]") {
+    GIVEN("Increasing ensemble size") {
+        int ens_size = GENERATE(10, 100, 1000);
+        int obs_size = 2;
+
+        std::vector<model> ens;
+
+        analysis_module_type *ies_module = analysis_module_alloc("IES_ENKF");
+
+        bool_vector_type *ens_mask = bool_vector_alloc(ens_size, true);
+        obs_data_type *obs_data = obs_data_alloc(1.0);
+        meas_data_type *meas_data = meas_data_alloc(ens_mask);
+        rng_type *rng = rng_alloc(MZRAN, INIT_DEFAULT);
+
+        model true_model{a_true, b_true};
+
+        for (int iens = 0; iens < ens_size; iens++)
+            ens.push_back(model(rng));
+
+        const char *obs_key = "OBS1";
+        meas_block_type *mb =
+            meas_data_add_block(meas_data, obs_key, 1, obs_size);
+        obs_block_type *ob =
+            obs_data_add_block(obs_data, obs_key, obs_size, nullptr, false);
+
+        for (int iobs = 0; iobs < obs_size; iobs++) {
+            double x = iobs;
+            for (int iens = 0; iens < ens_size; iens++) {
+                const auto &m = ens[iens];
+                meas_block_iset(mb, iens, iobs, m.eval(x));
+            }
+
+            obs_block_iset(ob, iobs, true_model.eval(x), 0.25);
+        }
+
+        matrix_type *E = obs_data_allocE(obs_data, rng, ens_size);
+
+        int nparam = 2;
+        matrix_type *A = matrix_alloc(nparam, ens_size); // prior
+        for (int iens = 0; iens < ens_size; iens++) {
+            matrix_iset(A, 0, iens, enkf_util_rand_uniform(0, 10, rng));
+            matrix_iset(A, 1, iens, enkf_util_rand_uniform(0, 1, rng));
+        }
+
+        std::unordered_map<std::string, matrix_type *> parameters = {{"A", A}};
+
+        WHEN("Running update without rowscaling") {
+            analysis::run_analysis_update_without_rowscaling(
+                ies_module, ens_mask, meas_data, obs_data, rng, E, parameters);
+
+            // The goal here is to show that the accuracy of estimation inscreases
+            // with increasing ensemble size.
+            THEN("The average of the parameter distribution should be close to "
+                 "the true parameter value") {
+                if (ens_size == 10) {
+                    REQUIRE(std::abs(a_true - matrix_get_row_sum(A, 0) /
+                                                  ens_size) < 0.9);
+                    REQUIRE(std::abs(b_true - matrix_get_row_sum(A, 1) /
+                                                  ens_size) < 0.1);
+                } else if (ens_size == 100) {
+                    REQUIRE(std::abs(a_true - matrix_get_row_sum(A, 0) /
+                                                  ens_size) < 0.12);
+                    REQUIRE(std::abs(b_true - matrix_get_row_sum(A, 1) /
+                                                  ens_size) < 0.015);
+                } else {
+                    REQUIRE(std::abs(a_true - matrix_get_row_sum(A, 0) /
```

Comment:
No change in values when going from `ens_size == 100` to `ens_size == 1000`?

Reply:
Since the numerical values used in the comparisons are tightly bound the ensemble sizes I would suggest using `else if (ens_size == 1000)` for the final check - instead of relying on a final catch all `else`